pandas/io/formats/format.py

```diff
@@ -1855,29 +1856,51 @@ def just(x):
     return result
 
 
-def _trim_zeros_complex(
-    str_complexes: np.ndarray, decimal: str = ".", na_rep: str = "NaN"
-) -> List[str]:
+def _trim_zeros_complex(str_complexes: np.ndarray, decimal: str = ".") -> List[str]:
     """
     Separates the real and imaginary parts from the complex number, and
     executes the _trim_zeros_float method on each of those.
     """
     return [
-        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal, na_rep))
+        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal))
         for x in str_complexes
     ]
 
 
+def _post_process_complex(complex_strings: List[str]) -> List[str]:
+    """
+    Zero pad complex number strings produced by _trim_zeros_complex.
+
+    Input strings are assumed to be of the form " 1.000+1.000j" with a
+    leading space and the real and imaginary parts having equal precision.
+    The output strings will all have length equal to that of the longest
+    input string.
+    """
+    lengths = [len(s) for s in complex_strings]
+    max_length = max(lengths)
+    padded = [
+        s[: -((k - 1) // 2 + 1)]  # real part
+        + (max_length - k) // 2 * "0"
+        + s[-((k - 1) // 2 + 1) : -((k - 1) // 2)]  # + / -
+        + s[-((k - 1) // 2) : -1]  # imaginary part
+        + (max_length - k) // 2 * "0"
+        + s[-1]
+        for s, k in zip(complex_strings, lengths)
+    ]
+    return padded
+
+
 def _trim_zeros_float(
-    str_floats: Union[np.ndarray, List[str]], decimal: str = ".", na_rep: str = "NaN"
+    str_floats: Union[np.ndarray, List[str]], decimal: str = "."
 ) -> List[str]:
     """
     Trims zeros, leaving just one before the decimal points if need be.
     """
     trimmed = str_floats
+    number_regex = re.compile(fr"\s*[\+-]?[0-9]+(\{decimal}[0-9]*)?")
```

Comment:
put this as a module level variable

Reply:
you may need a dict that is keyed by decimal; i am worried about perf here (or alt show that this not an issue)

Reply:
I think this is okay performance wise, I've run the io and string benchmarks a few more times each with better hardware and it's showing no significant changes.